src/cpl/mct/lnd_import_export.F90

```diff
@@ -136,6 +135,14 @@ real(r8) :: tdc, t               ! Kelvins to Celcius function and its input
        forc_rainl                                    = x2l(index_x2l_Faxa_rainl,i)   ! mm/s
        forc_snowc                                    = x2l(index_x2l_Faxa_snowc,i)   ! mm/s
        forc_snowl                                    = x2l(index_x2l_Faxa_snowl,i)   ! mm/s
+       ! tcraig, check for negative values as this causing problems in CTSM soil
+       ! only set to zero if they are very small negative values, otherwise let the code fail
+       ! typical max values of precip are 1.0e-4, set prec_limit to several orders of magnitude smaller
+       prec_limit = -1.0e-16
```

Comment:
I am just curious whether there is any particular reason to use this number instead of -1e-12 or other number.

Reply:
Sorry for the delay.  The number 1.0e-16 was relatively arbitrary.  I saw numbers around -1e-25 or so which is 20 order of magnitude less than the maximum precip value.  So I wanted to allow for some more negative values but significantly less than typical values so I picked -1e-16.  I could have just zeroed out all negative values, but I thought for now, we should just trap "tiny" negative values but allow larger ones thru so we could trap them.  In fact, maybe we should add a trap after these checks so if negative precip is getting thru it leads to an explicit model abort on the coupling field.

Reply:
No problem! That makes perfect sense to me!

Reply:
I think your idea @apcraig of trapping more negative precipitation values, printing a descriptive message and then aborting is good.  From talking with other WRF developers it seems like our very small negative precipitation is a bug in one (possibly more) microphysics schemes.  In the end, very small negative values should not kill the run, so zeroing them out as you're doing is perfect, but more negative values could indicate a problem